src/base/settingsstorage.h

```diff
@@ -62,4 +62,34 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
+    explicit StorableValue(const std::string &keyName, const QVariant &defaultValue = QVariant())
```

Comment:
don't know if this make sense:
if I wrote `explicit StorableValue(const std::string &keyName, const T &defaultValue = T())`
the I'll have to worry about if `T()` exists or not.